### ISSUE TITLE
IE8 support is broken

### ISSUE THREAD

**Issue by zapodot** [Reactive-Extensions/RxJS#1057]
Hi! Some of our users still run on IE8 and after introducing RxJS we have received an issue whlie including rx.lite.compat.js (v.4.0.7):

IE reports "stringClass" is undefined at line 236 which contains the following code 
```javascript
        self = splitString && toString.call(this) === stringClass ?
            this.split('') :
            object,
```
![image](https://cloud.githubusercontent.com/assets/551569/11745279/7cfd7b52-a017-11e5-8bb7-43723d904255.png)

Are there any workarounds?

**Comment by zapodot** [comment]
After adding 
```javascript
 var stringClass = '[object String]';
```
IE8 will complain about to other missing declartions, namely _funcClass_ and _arrayClass_.

Adding the following declarations fixes all errors in IE8:
```javascript
var arrayClass = '[object Array]',
      funcClass = '[object Function]',
      stringClass = '[object String]';
```

**Comment by mattpodwysocki** [comment]
Closed by #1060
